internal/dispatch/caching/caching.go

```diff
@@ -164,8 +165,11 @@ func (cd *CachingDispatcher) DispatchCheck(ctx context.Context, req *v1.Dispatch
 
 	// We only want to cache the result if there was no error
 	if err == nil {
-		toCache := checkResultEntry{computed, computed.Metadata.DepthRequired}
-		toCache.result.Metadata.DispatchCount = 0
+		adjustedComputed := proto.Clone(computed).(*v1.DispatchCheckResponse)
+		adjustedComputed.Metadata.CachedDispatchCount = adjustedComputed.Metadata.DispatchCount
+		adjustedComputed.Metadata.DispatchCount = 0
```

Comment:
Either here or in the cluster dispatcher we want to set the dispatch count for things that are server side cached to `1`, vs `0` if it was in the client-side cache.

Reply:
Probably best to be adjusted in the cluster dispatcher... although that would require cloning the message again, which kind of sucks.

Reply:
@jakedt let me know which you prefer

Reply:
Dispatcher would seem "right" to me, but if we want the responses to match for local dispatch vs remote dispatch, maybe the cache entries should just default to `1` always.

Reply:
So just set it to `1` here for dispatched, and always return that?

Reply:
Does the rationale seem sound to you?

Reply:
I believe so: If the local dispatch hits the cache, it has dispatched technically once, so it probably should return `1` from there.